sdk/metric/internal/aggregate/lastvalue.go

```diff
@@ -5,117 +5,179 @@ package aggregate // import "go.opentelemetry.io/otel/sdk/metric/internal/aggreg
+	newRes func(attribute.Set) FilteredExemplarReservoir[N]
+	values limitedSyncMap
+}
+
+func (s *lastValueMap[N]) measure(
+	ctx context.Context,
+	value N,
+	fltrAttr attribute.Set,
+	droppedAttr []attribute.KeyValue,
+) {
+	lv := s.values.LoadOrStoreAttr(fltrAttr, func(attr attribute.Set) any {
+		return &lastValuePoint[N]{
+			res:   s.newRes(attr),
+			attrs: attr,
+		}
+	}).(*lastValuePoint[N])
+
+	lv.value.Store(value)
+	lv.res.Offer(ctx, value, droppedAttr)
+}
+
+func newDeltaLastValue[N int64 | float64](
+	limit int,
+	r func(attribute.Set) FilteredExemplarReservoir[N],
+) *deltaLastValue[N] {
+	return &deltaLastValue[N]{
 		newRes: r,
-		limit:  newLimiter[datapoint[N]](limit),
-		values: make(map[attribute.Distinct]*datapoint[N]),
 		start:  now(),
+		hotColdValMap: [2]lastValueMap[N]{
+			{
+				values: limitedSyncMap{aggLimit: limit},
+				newRes: r,
+			},
+			{
+				values: limitedSyncMap{aggLimit: limit},
+				newRes: r,
+			},
+		},
 	}
 }
 
-// lastValue summarizes a set of measurements as the last one made.
-type lastValue[N int64 | float64] struct {
-	sync.Mutex
-
+// deltaLastValue summarizes a set of measurements as the last one made.
+type deltaLastValue[N int64 | float64] struct {
 	newRes func(attribute.Set) FilteredExemplarReservoir[N]
-	limit  limiter[datapoint[N]]
-	values map[attribute.Distinct]*datapoint[N]
 	start  time.Time
-}
 
-func (s *lastValue[N]) measure(ctx context.Context, value N, fltrAttr attribute.Set, droppedAttr []attribute.KeyValue) {
-	s.Lock()
-	defer s.Unlock()
-
-	d, ok := s.values[fltrAttr.Equivalent()]
-	if !ok {
-		fltrAttr = s.limit.Attributes(fltrAttr, s.values)
-		d = &datapoint[N]{
-			res:   s.newRes(fltrAttr),
-			attrs: fltrAttr,
-		}
-	}
-
-	d.value = value
-	d.res.Offer(ctx, value, droppedAttr)
+	hcwg          hotColdWaitGroup
+	hotColdValMap [2]lastValueMap[N]
+}
 
-	s.values[fltrAttr.Equivalent()] = d
+func (s *deltaLastValue[N]) measure(
+	ctx context.Context,
+	value N,
+	fltrAttr attribute.Set,
+	droppedAttr []attribute.KeyValue,
+) {
+	hotIdx := s.hcwg.start()
+	defer s.hcwg.done(hotIdx)
+	s.hotColdValMap[hotIdx].measure(ctx, value, fltrAttr, droppedAttr)
 }
 
-func (s *lastValue[N]) delta(
+func (s *deltaLastValue[N]) collect(
 	dest *metricdata.Aggregation, //nolint:gocritic // The pointer is needed for the ComputeAggregation interface
 ) int {
 	t := now()
+	n := s.copyAndClearDpts(dest, t)
+	// Update start time for delta temporality.
+	s.start = t
+	return n
+}
+
+// copyAndClearDpts copies the lastValuePoints held by s into dest. The number of lastValuePoints
+// copied is returned.
+func (s *deltaLastValue[N]) copyAndClearDpts(
+	dest *metricdata.Aggregation, //nolint:gocritic // The pointer is needed for the ComputeAggregation interface
+	t time.Time,
+) int {
 	// Ignore if dest is not a metricdata.Gauge. The chance for memory reuse of
-	// the DataPoints is missed (better luck next time).
+	// the lastValuePoints is missed (better luck next time).
```

Comment:
```suggestion
	// the DataPoints is missed (better luck next time).
```

This is still referring to the `dest` field.